packages/use-segment/src/useSegment.tsx

```diff
@@ -52,34 +54,49 @@ function SegmentProvider<T extends Events>({
   children,
   settings,
   initOptions,
+  areOptionsLoading = false,
   onError,
   onEventError,
   events,
 }: SegmentProviderProps<T>) {
+  const [isAnalyticsReady, setIsAnalyticsReady] = useState(false)
   const [internalAnalytics, setAnalytics] = useState<Analytics | undefined>(
     undefined,
   )
 
   const shouldLoad = useMemo(() => {
+    // If options are loading (for example Segment integrations),
+    // we don't know if we should load or not
+    if (areOptionsLoading) {
+      return undefined
+    }
+
     const hasNoIntegrationsSettings = !initOptions?.integrations
-    const isAllEnabled = !!initOptions?.integrations?.All
```

Comment:
I think it's still use full and should not be removed, as we can approved All integration do you know if one of the integration will be set to True ? if not isAnyIntegrationEnabled will be false also and nothing be load, just check in case it's happen